### DISCUSSION TITLE
Slice pattern got an error after add middleware (Typescript)

### DISCUSSION THREAD

**Discussion by detautama** [pmndrs/zustand#1176] [Q&A]
## Spec: 
- Zustand v4
- Nextjs 12.1.6
-  Typescript 4.6.4

## What I did: 
I have created a "slice pattern" for my store and added devtools middleware on them

## What I got: 
I got an error when trying to add "mutators reference". See image below
<img width="1040" alt="image" src="https://user-images.githubusercontent.com/18025983/183331042-5d7bbc3e-af53-4723-9555-f8246ddc92dd.png">

## What I expect: 
It should be no error on it, I have followed the doc but still got error. Link to doc: https://github.com/pmndrs/zustand/blob/main/docs/typescript.md#middlewares-and-their-mutators-reference
<img width="1045" alt="image" src="https://user-images.githubusercontent.com/18025983/183331226-c7bacb72-fb7b-4063-9375-50b291b2cff6.png">

Did I miss something? 

Thank you!

**Comment by dai-shi** [comment]
Can you create a reproduction with typescript playground?

**Reply by detautama** [reply]
Sure, I can't find any online editor like stackblitz or codesandbox with template Nextjs + Typescript. So I created a repository
https://github.com/detautama/gh-ds-1176-zustand/blob/main/store.ts

Thank you for your fast response

**Reply by dai-shi** [reply]
Okay, thanks.
I just copy&pasted to the playground.
https://tsplay.dev/wOGMdw
Does it show your issue well?

**Reply by dai-shi** [reply]
And, this is the fix.
https://tsplay.dev/mLq7VW

So, it looks like your issue is something different.

**Reply by detautama** [reply]
> Okay, thanks. I just copy&pasted to the playground. https://tsplay.dev/wOGMdw Does it show your issue well?

Your playground has 1 issue, but my playground has 2 issues. One issue is same
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/18025983/183337740-4ce0e873-ac9d-4d99-b2ec-bc207d74cf81.png">

**Reply by dai-shi** [reply]
Can you somehow reproduce it with [TypeScript Playground](https://www.typescriptlang.org/play)?

Maybe, TS version and/or tsconfig matter.

**Reply by detautama** [reply]
> And, this is the fix. https://tsplay.dev/mLq7VW
> 
> So, it looks like your issue is something different.

Ok, I know the problem. I missing to add the 'devtools' part in bound store

```
 create<BearSlice & FishSlice>()(
  devtools((...a) => ({
    ...createBearSlice(...a),
    ...createFishSlice(...a),
  }))
);
```


I think we need to add an example for adding middleware for slice pattern
I have created a PR for that :)
https://github.com/pmndrs/zustand/pull/1177

Thanks for your help!
